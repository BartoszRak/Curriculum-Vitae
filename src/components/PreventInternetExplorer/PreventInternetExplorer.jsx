import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { withStyles, Typography, Button } from '@material-ui/core'

import style from './PreventInternetExplorer.style'

export function PreventInternetExplorer({ classes, children }) {
  const { userAgent } = window.navigator
  const msie = userAgent.indexOf('MSIE ')

  const [isExplorer, setInternetExplorer] = useState(
    Boolean(msie > 0 || userAgent.match(/Trident.*rv:11\./))
  )
  return isExplorer ? (
    <div className={classes.root}>
      <Typography variant="h1">Oops!</Typography>
      <Typography variant="h2">Internet explorer detected!</Typography>
      <Typography variant="body1">
        Internet Explorer web browser in any version is not supported. Not because I
        am lazy devloper (because I am not) - it is not supported due to security
        reasons. Internet Explorer is outdated, old, based on old/unsecure
        technologies and there is not recommended to use it.
      </Typography>
      <Button
        className={classes.button}
        onClick={() => setInternetExplorer(false)}
        variant="contained"
      >
        Go unsecure
      </Button>
      <Typography variant="body2">
        * You see this warning because whole website or component should not be used
        on Internet Explorer.
      </Typography>
      <Typography variant="body2">
        ** Going unsecure may be dangerous, also component or website can behave
        improperly.
      </Typography>
    </div>
  ) : (
    children
  )
}

PreventInternetExplorer.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  classes: PropTypes.object,
}

PreventInternetExplorer.defaultProps = {
  children: null,
}

export default withStyles(style)(PreventInternetExplorer)
