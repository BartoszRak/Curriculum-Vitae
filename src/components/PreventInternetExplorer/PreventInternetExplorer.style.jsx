export default ({ spacing }) => ({
  button: {
    fontWeight: 'normal',
    marginBottom: spacing(1),
    marginTop: spacing(1),
  },
  root: {
    padding: spacing(1),
  },
})
