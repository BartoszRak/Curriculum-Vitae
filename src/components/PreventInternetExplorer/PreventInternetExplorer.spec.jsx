import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { PreventInternetExplorer } from './PreventInternetExplorer'

jest.unmock('@material-ui/core')

describe('component: PreventInternetExplorer', () => {
  afterEach(cleanup)

  describe('rendering', () => {
    test('match snapshot when loading', () => {
      const wrapper = render(<PreventInternetExplorer />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
