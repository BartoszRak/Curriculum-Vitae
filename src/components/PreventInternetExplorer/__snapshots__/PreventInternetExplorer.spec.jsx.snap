// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: PreventInternetExplorer rendering should render as expected 1`] = `
<div
  id="testChildren"
>
  Test children
</div>
`;
